### ISSUE TITLE
Error Message Is Not Announced

### ISSUE THREAD

**Issue by MajoMaldonado** [equalizedigital/accessibility-checker#1758]
Location / URL: https://edac-vpat.instawp.co/wp-admin/admin.php?page=accessibility_checker_settings&tab=accessibility-reports 
Screenshot: 

<img width="771" height="497" alt="Image" src="https://github.com/user-attachments/assets/70cfb4df-9d42-4791-a33a-9703eb596159" />

WCAG S.C: 4.1.3 Status Messages (Level AA)

Issue Description: When an invalid license key is entered, an error message is displayed visually but is not announced by screen readers when it appears. Error status messages should be programmatically identified so assistive technologies automatically announce them without requiring users to move focus or navigate to the message.

**Comment by pattonwebz** [comment]
@MajoMaldonado If this triggers a context change by the page refresh does 4.1.3 still apply? Is there a different pattern to follow on page refreshes?

**Comment by MajoMaldonado** [comment]
@pattonwebz The page does not refresh, I see there is still the same URL when you trigger the error. It is applicable.

**Comment by amberhinds** [comment]
I think it is an issue caused by how WordPress core handles saves, which is with a page reload rather than with ajax. The real question is whether we can make our settings pages save changes with ajax rather than using the reload behavior. 

I believe that @pattonwebz is right that this is not technically a 4.1.3 failure, because that criterion specifically addresses scenarios where new content is added to the page without changing the user's context, and because of the page reload there is a change of context.

That said, I do think we should leave this open, because it would be best for these messages to be announced.

**Comment by pattonwebz** [comment]
I asked about the SC number just for my own understanding here. I do still think this is a problem we can solve and I have solved this in other plugins by handling an announcement and optionally a focus shift on page reload. We can do that here too or we can swap to a no-reload-type-save and announce it as soon as save completes without a page refresh.

There's a solution here for sure, since there are a few open issues about similar problems we should find one solution that can fix all of them (and that can be used across other plugins as well which likely have similar patterns).
